sdk/internal/x/README.md

```diff
@@ -0,0 +1,46 @@
+# Experimental Features
+
+The SDK contains features that have not yet stabilized in the OpenTelemetry specification.
+These features are added to the OpenTelemetry Go SDK prior to stabilization in the specification so that users can start experimenting with them and provide feedback.
+
+These feature may change in backwards incompatible ways as feedback is applied.
+See the [Compatibility and Stability](#compatibility-and-stability) section for more information.
+
+## Features
+
+- [Resource](#resource)
+
+### Resource
+
+[OpenTelemetry resource semantic conventions] include many attribute definitions that are defined as experimental.
+To have experimental semantic conventions be added by [resource detectors] set the `OTEL_GO_X_RESOURCE` environment variable.
+The value set must be the case-insensitive string of `"true"` to enable the feature.
+All other values are ignored.
+
+<!-- TODO: document what attributes are added by which detector -->
+
+[OpenTelemetry resource semantic conventions]: https://opentelemetry.io/docs/specs/semconv/resource/
+[resource detectors]: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/resource#Detector
+
+#### Examples
+
+Enable experimental resource semantic conventions.
+
+```console
+export OTEL_GO_X_RESOURCE=true
```

Comment:
Should there be this single environment variable which turns on and off _all_ experimental semantic conventions, or is it the intention to have something more specific? For example, have a `OTEL_GO_X_RESOURCE_SERVICE`, which enables experimental attributes on the service resource?

Reply:
I don't expect there is a use case where a user wants some but not all of the experimental environment variables. 

If we find that to exist, we could add specific variables to support that then.

What do you think?

Reply:
Sounds good. Once this is merged, I'll submit a PR for `service.instance.id` guarded by the `Resource` flag.